pkgs/os-specific/linux/esdm/default.nix

```diff
@@ -0,0 +1,81 @@
+{ lib
+, stdenv
+, fetchzip
+, protobufc
+, pkg-config
+, fuse3
+, meson
+, ninja
+, libselinux
+, jitterentropy
+  # A more detailed explaination of the following meson build options can be found
+  # in the source code of esdm.
+  # A brief explanation is given:
+, selinux ? false # meson build option for selinux support
+, drng_hash_drbg ? true  # meson build option for setting the default drng callback
+, drng_chacha20 ? false # meson build option for setting the default drng callback
+, ais2031 ? true # meson build option for setting the seeding strategy to be compliant with AIS 20/31
+, linux-devfiles ? true # meson build option to enable linux /dev/random and /dev/urandom support
+, linux-getrandom ? true # meson build option to enable linux getrandom support
+, es_jitterRng ? true # meson build option to enable support for the entropy source: jitter rng
+, es_cpu ? true # meson build option to enable support for the entropy source: cpu-based entropy
+, es_kernel ? true # meson build option to enable support for the entropy source: kernel-based entropy
+, es_irq ? true # meson build option to enable support for the entropy source: interrupt-based entropy
+, es_sched ? true # meson build option to enable support for the entropy source: scheduler-based entropy
+, es_hwrand ? true # meson build option to enable support for the entropy source: /dev/hwrng
+, hash_sha512 ? false # meson build option for the conditioning hash: SHA2-512
+, hash_sha3_512 ? true # meson build option for the conditioning hash: SHA3-512
+}:
+
+assert drng_hash_drbg != drng_chacha20;
+assert hash_sha512 != hash_sha3_512;
+
+stdenv.mkDerivation rec {
+  pname = "esdm";
+  version = "0.6.0";
+
+  src = fetchzip {
```

Comment:
Since the upstream is on GitHub, you should use `fetchFromGitHub`. This will also enable automatic updates via the r-ryantm bot when a new release is tagged on GitHub.

Reply:
I have changed it now back, so that it uses fetchFromGitHub.
Also `rev` now is changed to `rev="v${version}"`